feat: add Flappy Bird game feature

Devanshusharma2005 opened the pull request:
## What does this PR do?

Added flappy bird game feature to cal video. 
Added a new enableFlappyBirdGame field to the CalVideoSettings schema and model
Created a database migration for the new field
Added the toggle to enable/disable the game in the event type setup page
Created a new FlappyBirdGame component with minimize and close functionality
Integrated the game into the video waiting room component
Added translations for the new feature
Added automatic hiding of the game when participants join the meeting

## Visual Demo (For contributors especially)

A visual demonstration is strongly recommended, for both the original and new change **(video / image - any one)**.

#### Video Demo (if applicable):

- Show screen recordings of the issue or feature.
- Demonstrate how to reproduce the issue, the behavior before and after the change.

#### Image Demo (if applicable):

- Add side-by-side screenshots of the original and updated change.
- Highlight any significant change(s).

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Added a Flappy Bird game to the video waiting room, with an option to enable or disable it in event settings.

- **New Features**
  - Added a toggle in event type setup to enable the Flappy Bird game.
  - Integrated the game into the waiting room with minimize and close options.
  - Game hides automatically when participants join.
  - Added translations and database support for the new setting.

<!-- End of auto-generated description by cubic. -->

asadath1395 committed:
feat: Addons to bookings

Devanshusharma2005 committed:
feat: add Flappy Bird game feature

Devanshusharma2005 committed:
feat: enhance Flappy Bird game with loading error handling and retry functionality

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
fix : some typecheck fixes

Devanshusharma2005 committed:
Merge branch 'feat/flappy-bird-integration' of https://github.com/Devanshusharma2005/cal into feat/flappy-bird-integration

Devanshusharma2005 committed:
refactor: type-check fixes

Devanshusharma2005 committed:
fix : cubic fixes

Udit-takkar committed:
Merge branch 'main' into feat/flappy-bird-integration

Udit-takkar reviewed (changes requested):
1) There are two crosses in flappy bird window
2) Flappy bird game is also displayed here as well which is wrong 

![Image](https://github.com/user-attachments/assets/3810f2ff-87bf-4368-8a26-59081ef954c9)


~3) I think we should add a new custom tray button that would open this game instead of opening it automatically and close it only when user clicks on cross button~

Like here we have custom tray button "Cal.ai", "Record" etc.

![Image](https://github.com/user-attachments/assets/3080ad0d-de26-4304-9f33-15dd0db311fe)

Devanshusharma2005 committed:
some more fixes

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
some more fixes...

Devanshusharma2005 committed:
Merge branch 'feat/flappy-bird-integration' of https://github.com/Devanshusharma2005/cal into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Udit-takkar committed:
fix: type error

Udit-takkar committed:
chore: add missing

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Udit-takkar committed:
chore: add enableAutomaticRecordingForOrganizer as well

Udit-takkar committed:
chore: documentation change

Udit-takkar committed:
chore: update description

Udit-takkar committed:
chore: change it to string

Udit-takkar committed:
chore: revert doc

Udit-takkar committed:
chore: undo openapi

Udit-takkar committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge main into feat/flappy-bird-integration

- Resolved conflicts in 7 files:
  - apps/api/v2/swagger/documentation.json
  - apps/web/public/static/locales/en/common.json
  - docs/api-reference/v2/openapi.json
  - packages/features/eventtypes/components/locations/Locations.tsx
  - packages/features/eventtypes/lib/types.ts
  - packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts
  - packages/prisma/zod/custom/eventtype.ts

- Integrated Flappy Bird game feature with main branch video settings structure
- Added enableFlappyBirdGame to CalVideoSettings interface and component
- Preserved all main branch functionality while adding flappy bird integration
- Updated localization strings for flappy bird game feature
- Fixed sessionStorage import issue in NavigationItem.tsx

Devanshusharma2005 committed:
Fix duplicate properties in create.handler.ts

- Removed duplicate enableAutomaticRecordingForOrganizer
- Removed duplicate enableAutomaticTranscription
- Removed duplicate disableTranscriptionForGuests
- Removed duplicate disableTranscriptionForOrganizer
- This resolves TypeScript compilation errors from merge conflicts

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Devanshusharma2005 committed:
Merge branch 'main' into feat/flappy-bird-integration

Udit-takkar reviewed (commented):
Type checks are failing

Devanshusharma2005 commented:
> Type checks are failing

fixed.!

Devanshusharma2005 committed:
Fix CalVideoSettings repository to include enableFlappyBirdGame field

- Added enableFlappyBirdGame to createCalVideoSettings method type and data
- Fixed CalVideoSettings component to remove Badge from flappy bird toggle
- This resolves the 'something went wrong' error on event types page
- Enables proper creation and updating of event types with flappy bird settings

Udit-takkar reviewed (commented):
There are some bugs in this PR

keithwillcode commented:
Closing for now since stale + conflicts
